source/git-api.js

```diff
@@ -106,12 +108,21 @@ exports.registerApi = (env) => {
       return false;  // ignore files that are in .gitignore
     } else if (filename.endsWith(".lock")) {
       return false;
-    } else if (filename.indexOf(path.join(".git", "refs")) > -1) {
-      return true;   // trigger for all changes under refs
-    } else if (filename == path.join(".git", "HEAD")) {
-      return true;   // Explicitly return true for ".git/HEAD" for branch changes
+    } else if (filename.indexOf(path.join("", "refs")) > -1) {
+      // trigger for all changes under directory named refs
+      // NOTE: This check will catch some other refs directories, but we can't look for .git
+      // in case the git directory has been relocated
+      return true;
+    } else if (filename.endsWith("HEAD")) {
```

Comment:
Let's not do this as `HEAD` can be a git directory.  There is no reason to change this

Reply:
I thought the point if this if statement was that ungit needs to watch the HEAD file. Given that the git directory may not be named ".git" the current check isn't going to watch the HEAD file in those git directories.

Reply:
@jung-kim Does my reasoning make sense here? Or am I not understanding what this piece of code is trying to do?

Reply:
@jung-kim does this make sense?

Reply:
@jung-kim Does this make sense?

Reply:
No, two reasons:

1. in corrected git repo after resolved via `git rev-parse`, `.../.git/HEAD` is the file and it should exist.
2. This will capture any and all file with names that ends with `HEAD`.  it's a simple file watch so it maybe innocuous but it's not specific enough.

Reply:
However not all git directories are named ".git". This is the reason for this pull request. So `HEAD` won't be watched at all in these repositories. That's why I made the change to look at all `HEAD` files.

Reply:
`git rev-parse` that you are suggesting will give us the correct git repo with the `.git` folder.

Reply:
Okay, I will await your solution.